zaas-service/src/main/java/org/zowe/apiml/zaas/zaas/ZaasExceptionHandler.java

```diff
@@ -89,11 +109,104 @@ public ResponseEntity<ApiMessageView> handleTokenNotValidException(RuntimeExcept
+            .body(messageView);
+    }
+
+    @ExceptionHandler(NoHandlerFoundException.class)
+    public ResponseEntity<ApiMessageView> handleNoResourceFoundException(NoHandlerFoundException e) {
+        log.debug("Resource not found", e);
+        ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.common.notFound").mapToView();
+        return ResponseEntity
+            .status(HttpStatus.NOT_FOUND)
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(messageView);
+    }
+
+    @ExceptionHandler(HttpRequestMethodNotSupportedException.class)
+    public ResponseEntity<ApiMessageView> handleMethodNotAllowedException(HttpServletRequest request, HttpRequestMethodNotSupportedException notAllowedMethodException) {
+        log.debug("MethodNotAllowedException exception", notAllowedMethodException);
+        ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.security.invalidMethod", request.getMethod(), request.getRequestURI()).mapToView();
+        return ResponseEntity
+            .status(HttpStatus.METHOD_NOT_ALLOWED)
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(messageView);
+    }
+
+    @ExceptionHandler(SSLException.class)
+    public ResponseEntity<ApiMessageView> handleSslException(HttpServletRequest request, SSLException sslException) {
+        log.debug("SSL exception", sslException);
+        ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.common.tlsError", request.getRequestURI(), sslException.getMessage()).mapToView();
+        return ResponseEntity
+            .status(HttpStatus.INTERNAL_SERVER_ERROR)
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(messageView);
+    }
+
+
+    @ExceptionHandler(UnsupportedResourceClassException.class)
+    public ResponseEntity<ApiMessageView> handleUnsupportedResourceClassException(UnsupportedResourceClassException unsupportedResourceClassException) {
+        log.debug("Unsupported resource class", unsupportedResourceClassException);
+        ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.security.common.auth.saf.endpoint.nonZoweClass", unsupportedResourceClassException.getResourceClass()).mapToView();
+        return ResponseEntity
+            .status(HttpStatus.INTERNAL_SERVER_ERROR)
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(messageView);
+    }
+
+    @ExceptionHandler(EndpointImproperlyConfigureException.class)
+    public ResponseEntity<ApiMessageView> handleEndpointImproprietyConfigureException(EndpointImproperlyConfigureException improprietyConfigureException) {
+        log.debug("Endpoint is improperly configured", improprietyConfigureException);
+        ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.security.common.auth.saf.endpoint.endpointImproprietyConfigure", improprietyConfigureException.getEndpoint()).mapToView();
```

Comment:
Also the key needs to be updated for the typo

```suggestion
        ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.security.common.auth.saf.endpoint.endpointImproperlyConfigured", improperlyConfiguredException.getEndpoint()).mapToView();
```